listener/listener.go

```diff
@@ -151,6 +153,16 @@ func WithSubscriptionDetails(lock time.Duration, maxDelivery int32) ManagementOp
 	}
 }
 
+func WithMessageLockAutoRenewal(interval time.Duration) Option {
+	return func(l *Listener) error {
+		if interval < time.Duration(0) {
+			return fmt.Errorf("renewal interval must be positive")
```

Comment:
what happens if I set it to 0? Should we have a lower limit of a second to stop sillyness?

Reply:
it will just keep looping on renewal, probably causing 429 really quick.
I think the flogging is equal to the silliness.